src/envoy/http/authn/authn_utils.cc

```diff
@@ -92,7 +90,7 @@ bool AuthnUtils::ProcessJwtPayload(const std::string& payload_str,
   ::google::protobuf::Map< ::std::string, ::std::string>* claims =
       payload->mutable_claims();
 
-  // Extract claims
+  // Extract claims as strings
```

Comment:
Since you are doing it for all claims, do you want to replace the logic here with what you had for ExtractStringList()?

Reply:
ExtractStringList() extracts a string list whereas the code here only extracts a string. So the logic for ExtractStringList() is not applied here.

Reply:
I mean, you can apply the same logic, first try as a string, then string list, here. With the current way, you actually do it twice, first cast all claims to string in this function, then cast them to string list in ExtractStringList function.

Reply:
Before  ExtractStringList(key, *json_obj, *claims, list) is called, claims must have been extracted. Hence the separation of two loops (one at line 93, the other at line 123).